clang/lib/AST/Type.cpp

```diff
@@ -2644,6 +2644,49 @@ bool QualType::isTriviallyCopyableType(const ASTContext &Context) const {
   return false;
 }
 
+bool QualType::isTriviallyCopyConstructibleType(
```

Comment:
Given that this function is exactly the same as the previous one, maybe we could
introduce a (non member static) `isTriviallyCopyableTypeImpl` function that checks either isTriviallyCopyableType or isTriviallyCopyConstructibleType , by either using a boolean or a callable, which isTriviallyCopyConstructibleType and isTriviallyCopyableType would dispatch to